api.php

```diff
@@ -834,42 +942,50 @@ public function load(ApiEntry $apiEntry) {
 	 * @throws ServerInternalException
 	 */
 	public function run() {
-		global $MC;
+		global $MC, $User;
 		
 		// Get route.
 		if (!isset($_GET['route']))
 			throw new RequestException('No route provided.');
-		$route = strtolower(trim($_GET['route']));
-		if (!isset($this->entries[$route]))
+
+		$this->route = strtolower(trim($_GET['route']));
+
+		if (!isset($this->entries[$this->route]))
 			throw new NotImplementedException('Unknown route.');
-		// Get user ID.
-		$apiKeyString = getBearerToken();
-		if ($apiKeyString == null)
-			throw new ClientUnauthorizedException('No API key provided.');
 
-		$key = str_replace(' ', '_', 'api' . $apiKeyString . $route );
-		if( false && $route == 'players/missing_orders' )
-		{
-			$result = $MC->get($key);
-			if ( $result )
-			{
-				return $result;
-			}
+		if ( !empty( $User ) && ( $User->type['User'] ?? false ) === true ){
+			/**
+			 * If the request is an API call using the existing user session, process using the ApiSession class. 
+			 */
+			$this->authClass = 'ApiSession';
+		}else{
+			/**
+			 * If the request is an API call using an API key, process using the ApiKey class. 
+			 */
+			$this->authClass = 'ApiKey';
 		}
 
+		$apiAuth = new $this->authClass($this->route);
 		// Get API entry.
-		$apiEntry = $this->entries[$route]; /** @var ApiEntry $apiEntry */
-		$apiKey = new ApiKey($apiKeyString);
+		$apiEntry = $this->entries[$this->route]; /** @var ApiEntry $apiEntry */
 		// Check if request is authorized.
-		$permissionIsExplicit = $apiKey->assertHasPermissionFor($apiEntry);
+		$permissionIsExplicit = $apiAuth->assertHasPermissionFor($apiEntry);
 		// Execute request.
-		$userID = $apiKey->getUserID();
+		$userID = $apiAuth->getUserID();
 		$result = $apiEntry->run($userID, $permissionIsExplicit); 
+		
+		// if( false && $route == 'players/missing_orders' )
+		// {
+		// 	$result = $MC->get($key);
+		// 	if ( $result )
+		// 	{
+		// 		return $result;
+		// 	}
+		// }
```

Comment:
remove added commented code

Reply:
I'm commenting this code because the if statement starts with `if( false )` which will always be false. However, I don't want to delete because it is their code.

Reply:
Are we sure that this isn't gonna affect anything on prod/staging?

Reply:
@abhishekCodazen yes, it start with `false` and uses an and statement. That code would not run in any programming language I know.